tests/storage-disks-vm

```diff
@@ -170,6 +170,96 @@ if [ "${DEVICE_HOTPLUG}" = "1" ]; then
+lxc config device add v2 block1rw disk source="${testRoot}/allowed1/lxd-block-test"
+lxc start v2
+waitInstanceReady v2
+
+lxc exec v2 -- mount | grep -F /srv/
+
+echo "==> Testing VM lxd-agent drive mounts"
+# Check there is only 1 mount for each directory disk and that it is mounted with the appropriate options.
+[ "$(lxc exec v2 -- mount | grep -cF '/srv/rw type')" = 1 ]
+[ "$(lxc exec v2 -- mount | grep -cF '/srv/ro type')" = 1 ]
+
+# RW disks should use virtio-fs when used with the snap.
+lxc exec v2 -- mount | grep 'lxd_dir1rw on /srv/rw type virtiofs (rw,relatime)'
+
+# RO disks should use virtio-fs when used with the snap but be mounted readonly.
+lxc exec v2 -- mount | grep 'lxd_dir1ro on /srv/ro type virtiofs (ro,relatime)'
+
+# Check UID/GID are correct.
+[ "$(lxc exec v2 -- stat -c '%u:%g' /srv/rw)" = '0:0' ]
+[ "$(lxc exec v2 -- stat -c '%u:%g' /srv/ro)" = '0:0' ]
+
+# Remount the readonly disk as rw inside VM and check that the disk is still readonly at the LXD layer.
+lxc exec v2 -- mount -oremount,rw /srv/ro
+lxc exec v2 -- mount | grep -F /srv/
+lxc exec v2 -- mount | grep 'lxd_dir1ro on /srv/ro type virtiofs (rw,relatime)'
+! lxc exec v2 -- touch /srv/ro/lxd-test-ro || false
+! lxc exec v2 -- mkdir /srv/ro/lxd-test-ro || false
+! lxc exec v2 -- rm /srv/ro/lxd-test.txt || false
+! lxc exec v2 -- chmod 777 /srv/ro || false
+
+## Mount the readonly disk as rw inside VM using 9p and check the disk is still readonly at the LXD layer.
+lxc exec v2 -- mkdir /srv/ro9p
+lxc exec v2 -- mount -t 9p lxd_dir1ro /srv/ro9p
+lxc exec v2 -- mount | grep -F /srv/
+lxc exec v2 -- mount | grep -x 'lxd_dir1ro on /srv/ro9p type 9p (rw,.*,access=client,trans=virtio)'
+! lxc exec v2 -- touch /srv/ro9p/lxd-test-ro || false
+! lxc exec v2 -- mkdir /srv/ro9p/lxd-test-ro || false
+! lxc exec v2 -- rm /srv/ro9p/lxd-test.txt || false
+! lxc exec v2 -- chmod 777 /srv/ro9p || false
+
+# Check writable disk is writable.
+lxc exec v2 -- touch /srv/rw/lxd-test-rw
+[ "$(stat -c '%u:%g' "${testRoot}/allowed1/lxd-block-test")" = "0:0" ]
+lxc exec v2 -- rm /srv/rw/lxd-test-rw
+lxc exec v2 -- rm /srv/rw/lxd-test
+
+# Check /dev/disk/by-id/ links
+lxc exec v2 -- test -b /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_block1ro
+lxc exec v2 -- test -b /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_block1rw
+
+# Check the rw driver accepts writes and the ro does not.
+! lxc exec v2 -- dd if=/dev/urandom of=/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_block1ro bs=512 count=2 || false
+lxc exec v2 -- dd if=/dev/urandom of=/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_block1rw bs=512 count=2
+
+echo "==> Checking disk device hotplug support"
+lxc config device add v2 block1 disk source="${testRoot}/allowed1/lxd-block-test" readonly=true
+# shellcheck disable=SC2016
+lxc exec v2 -- sh -c 'for i in $(seq 10); do test -e /sys/block/sdb/ro && break; echo "Waiting for sys file to appear (${i}s)"; sleep 1; done'
```

Comment:
Ah, another `sdb` showing up here and a few lines below. Would you mind deriving it from the destination of `/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_block1`.

Something like `DISK_DEV="$(basename "$(readlink -f /dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_lxd_block1)")"`